include/gz/math/graph/Edge.hh

```diff
@@ -204,9 +205,6 @@ namespace graph
   template<typename E>
   class UndirectedEdge : public Edge<E>
   {
-    /// \brief An invalid undirected edge.
-    public: static UndirectedEdge<E> NullEdge;
```

Comment:
We may choose to make a hard API break here, but I'd prefer to not remove these static types in this pull request, but to deprecate them instead, so that we can coordinate fixes for sdformat and any other `gz-*` packages in separate pull requests.

Reply:
Let me know if the changes in  [15b8975](https://github.com/gazebosim/gz-math/pull/606/commits/15b8975131bdef5f44c7932f4d120a15d21eca64) are sufficient to keep sdformat and friends compiling.

Reply:
it looks like sdformat15 will still compile with this change as of https://github.com/gazebosim/gz-math/pull/606/commits/5f606f0b6acfc11ec94ad077f7ea7229cd8ee44e (tested using the `ci_matching_branch/` trick partially documented in https://github.com/gazebosim/docs/pull/377)

[![Build Status](https://build.osrfoundation.org/buildStatus/icon?job=sdformat15-install_bottle-homebrew-amd64&build=210)](https://build.osrfoundation.org/view/gz-ionic/job/sdformat15-install_bottle-homebrew-amd64/210/) https://build.osrfoundation.org/view/gz-ionic/job/sdformat15-install_bottle-homebrew-amd64/210/